cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -53,6 +53,7 @@ var (
 	_                    = deprecatedflags.Add("connection-timeout")
 	volumeNamePrefix     = flag.String("volume-name-prefix", "pvc", "Prefix to apply to the name of a created volume.")
 	volumeNameUUIDLength = flag.Int("volume-name-uuid-length", -1, "Truncates generated UUID of a created volume to this length. Defaults behavior is to NOT truncate.")
+	volumeNamesReadable  = flag.Bool("volume-names-readable", false, "If enabled, includes the PVC namespace and name in VolumeRequests' suggested names.  Note that, combined with --volume-name-uuid-length, this can cause naming collisions.")
```

Comment:
Name collision can lead to stealing data. If attacker crafts namespace + pvc name that leads to collision with existing volume, CSI driver, obeying idempotency, must return the same volume as someone else is already using. Kubernetes is multi-tenant system and must not allow stealing data between namespaces.

Reply:
Yes; that is a trade off that must be considered when using this option.  Some of us are not administrating multi-tenant Kubernetes systems.  We're administering single-tenant Kubernetes systems that, for example, require an easy to way to find a given application's data in the storage provider.